pkg/controller/user/user.go

```diff
@@ -51,17 +51,28 @@ func (lc *userListController) ServeHTTP(w http.ResponseWriter, r *http.Request)
 	}
 	flash := flash.FromContext(w, r)
 
-	lc.logger.Infof("FLASH: %+v", flash)
+	realm, err := controller.GetRealm(w, r, user)
+	if err != nil {
+		flash.Error("Select realm to continue.")
+		http.Redirect(w, r, "/realm", http.StatusSeeOther)
+		return
+	}
 
 	m := html.GetTemplateMap(r)
 	m["user"] = user
+	m["realm"] = realm
 
-	users, err := lc.db.ListUsers(false)
-	if err != nil {
+	if err := realm.LoadRealmUsers(lc.db, false); err != nil {
```

Comment:
Same comment about authorized apps - I think it's fine for the realm to cache these, but I think it would be a cleaner API if this returned users:

```go
users, err := realm.GetUsers(...)
```

Reply:
Also a bit weird that it's called GetRealmUsers - you're already operating on a realm